p2p/conn/connection.go

```diff
@@ -249,9 +261,6 @@ func (c *MConnection) FlushStop() {
 		// so we dont race on calling sendSomePacketMsgs
 		<-c.doneSendRoutine
 
-		// Send and flush all pending msgs.
-		// By now, IsRunning == false,
```

Comment:
I deleted this comment because it is false. In my testing `IsRunning` was still true - that is why I needed to add the additional signaling. Maybe a better fix is to identify why `IsRunning` is still true.

Reply:
I looked closer and the reason is that FlushStop is outside of the `Stop` base service routine - this the base does not yet know that it is being stopped.